minesweeper.py

```diff
@@ -130,8 +130,11 @@ def timer(self):
                 self.game_time += datetime.timedelta(seconds=1)
                 root.after(1000, self.timer)
             elif self.game_status == GameStatus.game_won:
-                with open("high_scores.json", "w") as high_scores:
-                    json.dump(self.game_time.strftime("%M:%S"), high_scores)
+                with open(where_this_file_is / 'high_scores.json', "w") as score_list:
+                    json.dump({'time' : self.game_time.strftime("%M:%S")}, score_list)
+                    json.dump({'width' : self.width}, score_list)
+                    json.dump({'height' : self.height}, score_list)
+                    json.dump({'mine_count' : self.how_many_mines_user_wants}, score_list)
```

Comment:
When the game ends with win, it should append to `json_dict["high_scores"]`. That should go here (or even better, to the place where `self.game_status` is set to `GameStatus.game_won`).

The json writing code should go after `root.mainloop()`, so that it gets written after the gui window is closed. It should write `json_dict` to the file, not many weird little dicts.

The file contains more than just high scores, so maybe it should be renamed to `game_data.json` instead of `high_scores.json`.